doc/protocols/casa.rst

```diff
@@ -0,0 +1,93 @@
+
+The simplest solution would be to globally define the algorithm 
+(e.g. AES-based CMAC) that is used by such a protocol. If an 
+algorithm becomes obsolete, the protocol can be redefined using a 
+new Path Type, and another algorithm may be introduced. This 
+solution assumes that all the ASes in the whole Internet can agree 
+on the same cryptographic algorithms.
+
+However, ASes may want to be more flexible in the choice of the 
+algorithms they support. A more sophistic approach would be that 
+each AS promotes the subset of algorithms (from some globally 
+defined set of algorithms) that it supports, similarly to the 
+exchange of cipher suites in TLS handshakes. This means however, 
+that the data plane packets need to introduce an additional field 
+for each AS, which identifies the selected algorithm. For DRKey and 
+EPIC, border routers would need to potentially support many 
+different algorithms, which hinders scalability and performance.
+
+CASA can provide most of the advantages of both those two options, 
+while avoiding their drawbacks: it is a tradeoff between simplicity 
+and agility.
+
+Specification
+=============
+
+Description
+-----------
+In CASA, every AS chooses exactly **one algorithm** per algorithm 
+category. Those categories are globally defined and can either be 
+general (PRF, MAC, ...) or protocol-specific (DRKEY-PRF, 
+EPIC-MAC-HVF, EPIC-MAC-DVF, ...).
+
+::
+
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | MAC:          | CMAC-AES    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | PRF:          | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | DRKEY-PRF:    | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | EPIC-MAC-HVF: | Poly1305    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | EPIC-MAC-DVF: | Poly1305    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | ...           | ...         |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+The idea behind the general categories is to cover a large set of 
+protocols that only need algorithm agreement between different 
+entities.
+On the other hand, the protocol-specific categories serve the needs 
+of protocols that have additional requirements like high 
+performance. 
+
+Those algorithms are then primarily announced in the beacons, but 
+can also be exchanged in protocol-specific channels, for example 
+when ASes fetch each others first-level DRKeys.
+
+Beaconing Extensions
+--------------------
+For each category, the beacon needs to be extended by a field that 
+indicates the concrete algorithm implementation that an AS supports. 
+With a field size of one byte, there are 256 possible algorithms per 
+category.
+
+For a scenario with ten different categories and a path consisting 
+of eight ASes, the communication overhead introduced in the beacons 
+due to CASA amounts to 80 bytes.
+
+Advantages
+----------
+With CASA, every AS is free to choose which cryptographic algorithms 
+it wants to support, there is no global agreement necessary. 
```

Comment:
except for the algorithm categories

Reply:
... and the available options for the algorithms.

Reply:
Thank you, I added those points to the text.